### PR TITLE
Wip/hughsie/fu provider slowly

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — COMMENTED** [review]
Are you hard pressed on `provider`?  How about `interface_provider` or `hardware_interface`?

**Comment by hughsie** [comment]
> Are you hard pressed on provider? 

Nah, but I did spend 10 minutes with a thesaurus this morning.

> How about interface_provider or hardware_interface?

Both too wordy. I pondered FuSubsys but it's weird passing in `subsystems` _into_ a `FuUdevSubsys`. I guess the other logical one is `FuUsbBackend` but I'm not sure that's better. What I'm trying to convey is that it's a mechanism for how we get "this device has appeared" and "this device has gone away". `FuUsbInterface` is already overloaded and `FuUdevFacet` isn't super clear either. Maybe it's a factory. At the risk of sounding like Java, how about `FuUdevFactory` or FuUdevDeviceFactory? I'm not sure the metaphor holds 100% as a factory doesn't come and take the things it produces away again.

**Comment by superm1** [comment]
How about just `interface`?  or `hwif`?  Factory seems better than provider though too.

**Comment by superm1** [comment]
or `enumerator`?

**Comment by hughsie** [comment]
> or `enumerator`?

I think that's the verb, not the noun. Interface is going to cause all kinds of problems with GType. 😐

**Comment by hughsie** [comment]
FuUsbMethod?

**Comment by superm1** [comment]
I think backend and method are my top two for now then.

**Comment by hughsie** [comment]
I went with `backend` as it's better than provider.

**Comment by superm1** [comment]
OK LGTM

**Comment by hughsie** [comment]
I'll do some testing with the device-tests tomorrow and make sure we've not broken anything.
